cluster/gce/gci/append_or_replace_prefixed_line_test.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gci
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestAppendOrReplacePrefix(t *testing.T) {
+	testCases := []struct {
+		desc                string
+		prefix              string
+		suffix              string
+		initialFileContents string
+		want                string
+	}{
+		{
+			desc:   "simple string and empty file",
+			prefix: "hello",
+			suffix: "world",
+			want: `helloworld
+`,
+		},
+		{
+			desc:                "simple string and non empty file",
+			prefix:              "hello",
+			suffix:              "world",
+			initialFileContents: "jelloworld",
+			want: `jelloworld
+helloworld
+`,
+		},
+		{
+			desc:                "simple string and file already contains prefix",
+			prefix:              "hello",
+			suffix:              "world",
+			initialFileContents: "helloworld",
+			want: `helloworld
+`,
+		},
+		{
+			desc:   "string with quotes and $ and empty file",
+			prefix: "'\"$argon2id$v=19\"'",
+			suffix: "admin",
+			want: `"$argon2id$v=19"admin
+`,
+		},
+		{
+			desc:                "string with quotes and $ and non empty file",
+			prefix:              "'\"$argon2id$v=19\"'",
+			suffix:              "admin",
+			initialFileContents: "jelloworld",
+			want: `jelloworld
+"$argon2id$v=19"admin
+`,
+		},
+		{
+			desc:                "string with quotes and $ and file already contains prefix",
+			prefix:              "'\"$argon2id$v=19\"'",
+			suffix:              "admin",
+			initialFileContents: "\"$argon2id$v=19\"",
+			want: `"$argon2id$v=19"admin
+`,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
+			//Create an temp file
+			f, err := ioutil.TempFile("", "append_or_replace_test")
+			if err != nil {
+				t.Fatalf("Failed to create temp file.")
+			}
+			if err := ioutil.WriteFile(f.Name(), []byte(tc.initialFileContents), 0700); err != nil {
```

Comment:
`f.Write(tc.initialFileContents)`
and `f.Close()`

Reply:
Done.